datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -129,9 +164,13 @@ protected Duration _fromString(JsonParser parser, DeserializationContext ctxt,
         }
 
         try {
+            if (_durationUnitParser != null) {
+                return _durationUnitParser.parse(NumberInput.parseLong(value));
```

Comment:
Ok, here's another potential question: should JSON Strings be allowed with pattern?
In that case we'd probably want to check `CoercionConfig` as input absolutely must be (integer) number, so finding JSON String requires coercion. Same is not true for default `Duration` which conversely must get String.

I can probably play with this a bit so above is just my thinking out aloud.

Reply:
Only integers respects pattern for the moment.